test.py

```
@@ -0,0 +1,240 @@
+
+def test_extract_sections_from_txt(sample_text):
+    """
+    Tests that extract_sections_from_txt correctly extracts all sections from sample text.
+    Checks that the number of sections and first section header are as expected.
+    """
+    sections = extract_sections_from_txt(sample_text, exclude_sections=[])
+    assert len(sections) == 6
+    assert sections[0]["header"] == "introduction"
+    assert "overview of the document" in sections[0]["text"]
+
+
+def test_extract_sections_with_exclusion(sample_text):
+    """
+    Tests that extract_sections_from_txt excludes specified sections properly.
+    Verifies that excluded headers like 'references' and 'acknowledgements' are omitted.
+    """
+    sections = extract_sections_from_txt(
+        sample_text,
+        exclude_sections=["acknowledgement", "acknowledgements", "reference", "references"]
+    )
+    headers = [s["header"] for s in sections]
+    assert "references" not in headers
+    assert "acknowledgements" not in headers
+    assert len(sections) == 4
+
+
+# --- summarize_with_transformer tests ---
+
+def test_summarize_with_transformer(mock_summarizer):
+    """
+    Tests summarize_with_transformer using a mocked summarizer.
+    Ensures that the function returns the expected fixed summary for non-empty input.
+    """
+    with patch("summarization_script.pipeline", return_value=mock_summarizer):
+        text = "This is a long text that needs summarization."
+        summary = summarize_with_transformer(text)
+        assert summary == "This is a summary."
+
+
+def test_summarize_with_transformer_empty(mock_summarizer):
+    """
+    Tests summarize_with_transformer with empty input text.
+    Verifies that the mocked summarizer still returns the fixed summary.
+    """
+    with patch("summarization_script.pipeline", return_value=mock_summarizer):
+        text = ""
+        summary = summarize_with_transformer(text)
+        assert summary == "This is a summary."
```

```suggestion
        assert summary == "Error: Summarization failed."
```
This test keeps failing because, since here we are testing the summarizer on empty input, it should throw the exception - "Error: Summarization failed." rather than "This is a summary"